packages/dds/merge-tree/docs/Attribution.md

```diff
@@ -0,0 +1,378 @@
+
+At the time of writing this document, the Fluid framework doesn't natively support this kind of functionality,
+though in theory it has all of the data it needs (the op envelope contains both a timestamp and a client id, which can
+be mapped to information about the user using the audience).
+This has forced Fluid consumers that want attribution information to use workaround schemes. For example, in SharedString it's
+straightforward to conceptualize a scheme where each time a client submits an op that edits the string, it waits for that op to
+ack and uses the timestamp on that op to submit an additional op that annotates the edited segments with attribution information.
+
+Besides unnecessarily complicating client code, this has several drawbacks:
+- It is noisy on the wire
+- Attribution information can be lost in various cases if the submitting client disconnects
+- In-memory and snapshot size for the SharedString is more bloated than it should be; without binning the timestamps this strategy
+  entirely invalidated the zamboni scheme, and even if the timestamps are binned this will unnecessarily include the same user info
+  many times on different segments
+
+Rather than force this burden on consumers, it makes more sense to bake some attribution capability into the Fluid framework in an opt-in way.
+Though this document will cover an approach for doing so in merge-tree (primarily targeted at support for attribution in SharedString),
+none of the above concerns are specific to a single DDS.
+It's imaginible that Fluid will eventually want to generalize this to a platform mechanism that's supported by each DDS that wants to opt in to it.
+For that reason, the design is aimed to modularize into areas that are generic to the container runtime and those that are DDS-specific.
+
+## High-level
+
+If one had access to the entire op stream, a lookup from all historical client ids to their user info,
+and every DDS retained information about which sequence number created/modified each part of its data,
+attribution would be straightforward. Ask the DDS for the relevant sequence number, then look at this sequence number's op for a timestamp + clientId
+and use the client id to look up user information.
+
+All of this information is knowable from the fluid runtime perspective, though not all of it is persisted indefinitely.
+Notably:
+- Access to the entire op stream is an unreasonable assumption due to the summarization process
+- User information is only accessible for connected clients
+
+However, this conceptualization of attribution does suggest a reasonable split of concerns that can be individually assessed:
+none of the association between sequence numbers, timestamps, clientIds, and user information is specific to any given DDS.
+Thus, all of this bookkeeping could be generically done by the framework (potential candidates include on container runtime, data store runtime, or channel context),
```

Comment:
something interesting to consider in this regard, as that an individual op only ever targets a single entity. usually, the entity is a dds. non dds case are possible, but not very interesting for this discussion, so i'll use dds from here on out. since ops have a one to one mapping with a single dds, some of this bookkeeping could be pushed down to the dds itself in an opt in manner, for this discussion that interesting piece is probably just timestamp, but one could also image dds storing serialized/linearized ops (each op rebased to previous op), something like that would then allow time travel like functionality, which is out of scope here, but still interesting. This could also be useful when thinking about garbage collection, as the only data shared across dds will be user info, as the dds could internally manage sequence number to timestamp and anything else it wants. The easiest solution here would be to just put all the user data in the dds too, but that could cause duplications we may not want, so user data only could be lifted.